File: Add Code Here/PYTHON/1-two-sum/1-two-sum.js
// Test implementation

const twoSum = (nums, target) => {
    const hash = {};
    for (let i = 0; i < nums.length; i++) {
        const num = nums[i];
        const complement = target - num;
        if (hash[complement] !== undefined) {
            return [hash[complement], i];
        }
        hash[num] = i;
    }
    return [-1, -1];
}

File: Add Code Here/PYTHON/1-two-sum/1-two-sum.cpp
// Test implementation

#include <unordered_map>
#include <iostream> 
#include <vector>

using std::unordered_map;
using std::vector;


class Solution 
{
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
        unordered_map<int,int> hash;
        auto end= hash.end();
        vector<int> solution(2,0);
        
        for (int i=0;i<nums.size();i++)
        {
            
            auto find =hash.find(target-nums[i]);
            
            if(find==end)
            {   
                 hash.emplace(nums[i],i);   
            }
            else
                
            {
                solution={find->second,i};
                
                return solution;     
            }     
        }
        return solution;
    }
};